src/deploy/campaign/deploy-campaign.ts

```diff
@@ -0,0 +1,85 @@
+import { ICampaignArgs, ICampaignState, IDeployCampaignConfig } from "./types";
+import { Deployer } from "../deployer/deployer";
+import { TDeployMissionCtor } from "../missions/types";
+import { BaseDeployMission } from "../missions/base-deploy-mission";
+import { Contract } from "ethers";
+import { BaseStorageAdapter } from "../storage/base-storage-adapter";
+
+
+export class DeployCampaign {
+  state : ICampaignState;
+  deployer : Deployer;
+  // TODO dep: fix typing
+  dbAdapter : BaseStorageAdapter;
+  logger : Console;
+  config : IDeployCampaignConfig;
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  [name : string | symbol] : any;
+
+  private static indexedHandler : ProxyHandler<DeployCampaign> = {
+    get: (target, prop) => {
+      if (typeof prop === "string" && !!target.state.contracts[prop]) {
+        return target.state.contracts[prop];
+      } else if (typeof prop === "string" && !!target[prop]) {
+        return target[prop];
+      }
+
+      throw new Error(`
+      Accessing deployed contract data that is not in Campaign State or
+      property '${prop.toString()}' does not exist on DeployCampaign.
+      `);
+    },
+  };
+
+  constructor ({
+    missions,
+    deployer,
+    dbAdapter,
+    logger,
+    config,
+  } : ICampaignArgs) {
```

Comment:
If you've already made a type `ICampaignArgs` for the code, consider making the input object just a single variable

Reply:
no need to convert it to a single variable and then destructure or access again from it in the constructor. changing one destructuring for another here